src/audio/selector/selector.c

```diff
@@ -173,11 +173,8 @@ static int selector_params(struct comp_dev *dev,
 	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
 				source_list);
 
-	/* rewrite channels number for other components */
-	if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
-		sinkb->stream.channels = cd->config.out_channels_count;
-	else
-		sourceb->stream.channels = cd->config.in_channels_count;
+	sourceb->stream.channels = params->channels;
+	sinkb->stream.channels = cd->config.out_channels_count;
```

Comment:
As I understand correctly, in general we can configure selector with ```out_channels_count``` equal to 0 (which means that this value can vary). In such scenario we should not  do:
```sinkb->stream.channels = cd->config.out_channels_count;```
because it may cause some issues.

Reply:
> As I understand correctly, in general we can configure selector with `out_channels_count` equal to 0 (which means that this value can vary). In such scenario we should not do:

I never heard about this, then who decide the exact value for this "0" case?
Per my understanding, this out_channels_count can be either 1, or same to the in_channels_count only, @lbetlej @slawblauciak  @mrajwa should know about this.

> `sinkb->stream.channels = cd->config.out_channels_count;`
> because it may cause some issues.

Reply:
@bkokoszx @keyonjie can we make sure these assumptions are well documented in the code as part of this PR once you guys have aligned. + @mrajwa and @mmaka1 for their inputs.

Reply:
Ok, I will talk with Lech after his return.